### PR TITLE
Change fetch_breakpoints() to allow pending ones, if wanted

### PR DESCRIPTION
**Pull request by gavinhoward**
This PR also ensures that only valid breakpoints are fetched and adds a style attribute to the breakpoints module to show pending breakpoints.

You can feel free reject this PR; I just thought it might be a good idea.

### CODE REVIEW

File: .gdbinit [hunk 1]
```diff
@@ -270,6 +272,8 @@ def fetch_breakpoints(regular_only):
         breakpoint['condition'] = gdb_breakpoint.condition
         breakpoint['temporary'] = gdb_breakpoint.temporary
         breakpoint['hit_count'] = gdb_breakpoint.hit_count
+        breakpoint['pending'] = gdb_breakpoint.pending
+        breakpoint['visible'] = gdb_breakpoint.visible
```
Comment (cyrus-and) [inline .gdbinit]: Do you need the visibility status for your own purpose? Besides, can this ever be `False`? The documentation is a bit obscure IMHO:

> This attribute tells whether the breakpoint is visible to the user when set, or when the ‘info breakpoints’ command is run. This attribute is not writable.

File: .gdbinit [hunk 2]
```diff
@@ -258,6 +258,8 @@ def fetch_breakpoints(regular_only):
     # information
     breakpoints = []
     for gdb_breakpoint in gdb.breakpoints():
+        if not gdb_breakpoint.is_valid() or (not allow_pending and gdb_breakpoint.pending):
```
[the remark this answers is not in the document]
Comment (gavinhoward) [reply .gdbinit]: Hmm...I thought that there might be a `delete` command run and then `gdb-dashboard` updates, but the breakpoint object is still there. However, you know `gdb` more than I do, and I trust your judgment. Will fix.

File: .gdbinit (hunk 1 above)
Comment (gavinhoward) [reply .gdbinit]: Oops! That was an artifact that I meant to get rid of. (I put it in while working on the PR, but meant to take it out.) Will fix.

**Comment by gavinhoward** [comment]
All of your comments should be fixed now.

File: .gdbinit (hunk 2 above)
[the remark this answers is not in the document]
Comment (cyrus-and) [reply .gdbinit]: We're requesting a fresh new list via `gdb.breakpoints()` every time so that should never happen.

**Comment by cyrus-and** [comment]
> I'm not sure, showing the pending status is somewhat redundant as pending breakpoints are already distinguished by the fact that there is no actual location, e.g.:
> 
> ```
> [1] break for some_unknown_symbol
> ```

I'm still not sure about this...

**Comment by gavinhoward** [comment]
Oh, I fixed that one too. Sorry for not mentioning that.

**Comment by cyrus-and** [comment]
Hmm then what's remaining? Pending breakpoints are already returned by `fetch_breakpoints` and displayed in the module even though I don't check the `.pending` flag explicitly.

Is having the `breakpoint['pending']` field in the returned breakpoints for your own module development part of the goal of this PR?

**Comment by gavinhoward** [comment]
I have two goals:

1. I can turn off pending breakpoints in the `Breakpoints` module. (Sometimes, they get in the way.)
2. That I can query if breakpoints are pending (for another module I am making).

I apologize if this doesn't fit your goals for `gdb-dashboard`. I will not feel bad if you close this without merging. I am sure I can find another way to extend `gdb-dashboard` to do what I would like.

**Comment by cyrus-and** [comment]
Ahh, my bad, I thought you wanted to *include* pending breakpoints. Anyway, I'm OK with the PR but I've changed the attribute name to just `pending` and its default value to `True`.

**Comment by gavinhoward** [comment]
Not a bad idea to change that default. :) Thank you very much.

**Comment by cyrus-and** [comment]
You're welcome!

FYI 829e94a9ec497d90c74f2c03de4a5573dda916da changes the signature of `fetch_breakpoints`. I don't want to introduce unwanted bugs in your module. :)

```python
def fetch_breakpoints(watchpoints=False, pending=False):
```

**Comment by gavinhoward** [comment]
No worries, I didn't finish the module yet. :)

**Pull request merged**